src/live_restore/live_restore_fs.c

```diff
@@ -1207,109 +1067,97 @@ __wt_live_restore_fh_import_extents_from_string(
 
-    if (lr_fh->destination.hole_list_head != NULL) {
-        WT_ASSERT_ALWAYS(
-          session, extent_string_empty, "Extent list not empty while trying to import");
+    if (lr_fh->destination.bitmap != NULL) {
+        WT_ASSERT_ALWAYS(session, false, "Bitmap not empty while trying to import");
         return (0);
     }
 
-    if (extent_string_empty) {
-        if (lr_fh->destination.newly_created)
-            WT_ERR(__live_restore_alloc_extent(
-              session, 0, lr_fh->source_size, NULL, &lr_fh->destination.hole_list_head));
-        else
-            lr_fh->destination.complete = true;
-    } else {
-        __wt_readlock(session, &lr_fh->ext_lock);
-        __wt_verbose_debug3(session, WT_VERB_LIVE_RESTORE, "%s metadata extent list string: %s",
-          fh->name, extent_str);
-        /* The extents are separated by ;. And have the shape %d-%u. */
-        wt_off_t off = 0, next_off;
-        size_t len;
-        WTI_LIVE_RESTORE_HOLE_NODE **current = &lr_fh->destination.hole_list_head;
-        const char *str_ptr = extent_str;
-        char *next;
-        while (true) {
-            if (!__wt_isdigit((u_char)*str_ptr))
-                WT_ERR_MSG(session, EINVAL, "Invalid offset found in extent string");
-
-            next_off = (wt_off_t)strtoll(str_ptr, &next, 10);
-            str_ptr = next;
-            if (*str_ptr == '\0')
-                WT_ERR_MSG(session, EINVAL, "Invalid separator found in extent string");
-
-            /*
-             * Extents are additive to compress the string size i.e. the offset of extent n + 1 is
-             * the offset of extent n plus the offset of extent n + 1.
-             */
-            off += next_off;
-            str_ptr++;
-            if (!__wt_isdigit((u_char)*str_ptr))
-                WT_ERR_MSG(session, EINVAL, "Invalid length found in extent string");
-
-            len = (size_t)strtol(str_ptr, &next, 10);
-            if (len == 0)
-                WT_ERR_MSG(session, EINVAL, "Length zero extent found, this is an error");
-
-            __wt_verbose_debug3(session, WT_VERB_LIVE_RESTORE,
-              "Adding an extent: %" PRId64 "-%" WT_SIZET_FMT, off, len);
-            WT_ERR(__live_restore_alloc_extent(session, off, len, NULL, current));
-            current = &((*current)->next);
-
-            str_ptr = next;
-            /* We've reached the end of the string, don't go over by accident. */
-            if (*str_ptr == '\0')
-                break;
-            str_ptr++;
-        }
-        WT_ERR(__live_restore_handle_verify_hole_list(
-          session, (WTI_LIVE_RESTORE_FS *)S2C(session)->file_system, lr_fh, fh->name));
+    __wt_readlock(session, &lr_fh->bitmap_lock);
+    lr_fh->allocsize = lr_fh_meta->allocsize;
+    if (lr_fh->destination.newly_created) {
+        /*
+         * Only allocate a bitmap for a file that has been created in the destination for this run.
```

Comment:
```suggestion
         * Only allocate a bitmap for a file that has been newly created in the destination and has a backing source file.
```
What does "for this run" mean? 
I would also prefer this comment above the if statement.

Reply:
8b9922c0b

Reply:
I was thinking of "run" as iteration of WiredTiger existing.

Reply:
Hmm, run isn't clear but happy with this now!